### PR TITLE
Setup CI pipeline for Andela Socials

### PR DESCRIPTION
**Pull request by natalie-elizabeth**
#### What Does This PR Do?

Create a circle CI pipeline to run test after every commit.

#### Description of Task to be completed?
This task enables the application to run tests after a commit is made.

### CODE REVIEW

File: scripts/start.sh [hunk 1]
```diff
@@ -8,26 +8,11 @@ echo 'Enter y or n'
...
-  else
-    echo -e '\n\n\033[31mYou do not have a venv folder!\033[0m\n'
-  fi
+if [ $response == "n" ]; then
+  source venv/bin/activate
 else
   echo "Virtualenv Wrapper in use!"
 fi
 
-function start_client() {
```
Comment (andela-sjames) [inline scripts/start.sh]: Why are we deprecating this? Does this work outside docker? Also, how are you calling this script?

`scripts/start.sh` is how it is called via the readme. Will this still work if called like that, I just need some context here

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,119 @@
+
+PROJECT_NAME ?= andela-socials
```
Comment (andela-sjames) [inline Makefile]: Why do we have two separate makefiles?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	@ docker-compose -p $(DOCKER_TEST_PROJECT) -f $(DOCKER_TEST_COMPOSE_FILE) down -v
+	${INFO} "Removing dangling images..."
+	@ docker images -q -f dangling=true | xargs -I ARGS docker rmi -f ARGS
+	@ docker images -q -f dangling=true -f label=application=$(PROJECT_NAME) | xargs -I ARGS docker rmi -f ARGS
+	${SUCCESS} "Clean complete"
+
+
+## Delete local development server containers
+clean:
+	${INFO} "Cleaning your local environment, note all ephemeral volumes will be destroyed"
```
Comment (andela-sjames) [inline Makefile]: The usage of this makefile should also be added to the readme

File: docker/test/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+ENV CI=true
+
+RUN apt-get -y update && apt-get -y install xvfb libgtk2.0-0 libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2
+
+ADD scripts/setup.sh /usr/local/bin/setup.sh
+
+ADD scripts/start.sh /usr/local/bin/start.sh
+
+RUN chmod +x /usr/local/bin/setup.sh
```
Comment (andela-sjames) [inline docker/test/Dockerfile]: This is not necessary if we already have `chmod +x` set on the file locally.  That will reduce the image size at best.

File: Makefile (hunk 2 above)
Comment (natalie-elizabeth) [reply Makefile]: I hadn't realized both were pushed.

File: Makefile (hunk 3 above)
Comment (natalie-elizabeth) [reply Makefile]: Noted.

**Pull request merged**